File: q62_0912.py
import numpy as np

class Solution(object):
    def numTilings(self, n):
        """
        :type n: int
        :rtype: int
        """
        arr = np.zeros(n + 3, dtype=int)
        arr[0] = 1
        arr[1] = 2
        arr[2] = 5
        
        for i in range(3, n):
            arr[i] = 2 * arr[i-1] + arr[i-3]
            
        return arr[n-1] % 100000007
        
        

File: q62_0912.cpp
class Solution {
public:
    int numTilings(int n) {
        long* tile = new long [n];

        tile[0] = 1;

        if(n > 1){
            tile[1] = 2;
        }

        if(n > 2){
            tile[2] = 5;
        }

        for(int i = 3; i < n; i ++){
            tile[i] = (2 * tile[i - 1] + tile[i - 3] )% 1000000007;
        }

        return (int) tile[n-1];
    }
};